src/main/scala/org/deepdive/ddlog/DeepDiveLogParser.scala

```diff
@@ -23,6 +23,7 @@ case class NullConst extends ConstExpr
 case class FuncExpr(function: String, args: List[Expr], isAggregation: Boolean) extends Expr
 case class BinaryOpExpr(lhs: Expr, op: String, rhs: Expr) extends Expr
 case class TypecastExpr(lhs: Expr, rhs: String) extends Expr
+case class Placeholder extends Expr
```

Comment:
Placeholder is not an Expr.  You can't compile it to any SQL expression, or evaluate it.  We need a separate type hierarchy for patterns as I suggested earlier.  How about this:

```scala
sealed trait Patt
case class VarPatt(name: String) extends Patt
case class ExprPatt(expr: Expr) extends Patt
case class WildcardPatt extends Patt
```